api/src/orgs/data.js

```diff
@@ -39,6 +46,55 @@ export function getOrganisations (
   ).then(withOrgStats(query, { ...filter }))
 }
 
+export async function updateProfile (
+  db,
+  args
+) {
+  const { address, signerAddress, signedMessage, ...updateParams } = args
+
+  const originalMessage = composeSignedMessage(address, updateParams)
+  const isAddressValid = validateSignerAddress(originalMessage, signedMessage, signerAddress)
+  Web3EthContract.setProvider('wss://mainnet.eth.aragon.network/ws')
+
+  const kernelContract = new Web3EthContract(kernelAbi, address)
+
+  try {
+    const hasPermission = await kernelContract.methods.hasPermission(
+      signerAddress,
+      address,
+      MANAGE_PROFILE_ROLE,
+      EMPTY_SCRIPT
+    ).call()
+    if (!isAddressValid || !hasPermission) {
+      throw new Error('Failed message verification.')
+    }
+  } catch (err) {
+    console.error(err)
+    return
+  }
+
+  const { profile = {} } = await db.collection('orgs').findOne({ address })
+  const currentEditors = profile.editors || []
+  const newEditors = [...currentEditors, signerAddress]
+  // Remove duplicates
+  const filteredEditors = new Set(newEditors)
+  const newFilteredEditors = [...filteredEditors]
```

Comment:
I think this code would achieve the same result (we might want to rename `newFilteredEditors` to something more succinct in that case)

```suggestion
  const {
    profile = {}
  } = await db.collection('orgs').findOne({ address })
  const newFilteredEditors = new Set(profile.editors || [])
  newFilteredEditors.set(signerAddress)
```

Reply:
Way nicer! Done in https://github.com/1Hive/apiary/pull/181/commits/fc57eb6088b2c7cafe0bdce21a4b14c03f6523eb.